### DISCUSSION TITLE
Option for Suboscillator

### DISCUSSION THREAD

**Discussion by ErichHeinemann** [marcel-licence/esp32_basic_synth#34] [Ideas]
Hi Marcel, 
on old Synths, sometimes they provide the option to tune a oscillator an octave lower.
Based on the Voice/Oscillator - management, I did not find the right place to do so.

**Comment by marcel-licence** [comment]
Hi Erich,

in line 696 there will be setup an additional oscillator (one octave up). You can change this to one octave down or also copy the code to setup multiple osc per note:

```
    osc = getFreeOsc();
    if (osc != NULL)
    {
        if (note + 12 < 128)
        {
            osc->addVal = midi_note_to_add[note + 12];
            osc->samplePos = 0; /* we could add some offset maybe */
            osc->waveForm = &selectedWaveForm2;
            osc->dest = voice->lastSample;
            osc->pan_l = 1;
            osc->pan_r = 1;

            osc_act += 1;
        }
    }
```

**Comment by GillesLACAUD** [comment]
You can also set the sub to any semitone (between -24 to +12) so you can add some 5th sound (or any combination) just assign the +12 to a variable and control it with an Midi CC
